src/common/transformations/src/transformations/low_precision/mark_dequantization_subgraph.cpp

```diff
@@ -309,3 +309,56 @@ ov::pass::KeepConstPrecision::KeepConstPrecision(const element::TypeVector& prec
     auto m = std::make_shared<Matcher>(multiply_pattern, "KeepConstPrecision");
     this->register_matcher(m, callback);
 }
+
+ov::pass::KeepDequantizationPrecision::KeepDequantizationPrecision(const element::TypeVector& precisions) {
+    MATCHER_SCOPE(KeepDequantizationPrecision);
+
+    auto input_pattern = pattern::any_input(pattern::type_matches_any(precisions));
+    auto convert_pattern = pattern::wrap_type<v0::Convert>({input_pattern}, pattern::consumers_count(1));
+
+    // zero points:
+    auto zp_pattern = pattern::any_input();
+    auto zp_convert_pattern = pattern::optional<v0::Convert>(zp_pattern);
+    auto zp_reshape_pattern = pattern::optional<v1::Reshape, v0::Unsqueeze>({zp_convert_pattern, pattern::any_input()});
+    auto subtract_pattern = pattern::optional<v1::Subtract>({convert_pattern, zp_reshape_pattern});
+
+    // scale:
+    auto scale_pattern = pattern::any_input();
+    auto scale_convert_pattern = pattern::optional<v0::Convert>(scale_pattern);
+    auto scale_reshape_pattern =
+        pattern::optional<v1::Reshape, v0::Unsqueeze>({scale_convert_pattern, pattern::any_input()});
+    auto multiply_pattern = pattern::wrap_type<v1::Multiply>({subtract_pattern, scale_reshape_pattern});
+
+    matcher_pass_callback callback = [=](Matcher& m) {
+        const auto& pt_map = m.get_pattern_value_map();
+        auto convert = pt_map.at(convert_pattern).get_node_shared_ptr();
+        auto multiply = m.get_match_root();
+
+        if (transformation_callback(multiply)) {
+            return false;
+        }
+
+        disable_fp16_compression(convert);
+        disable_fp16_compression(multiply);
+
+        if (pt_map.count(subtract_pattern)) {
+            auto subtract = pt_map.at(subtract_pattern).get_node_shared_ptr();
+            disable_fp16_compression(subtract);
+        }
+
+        if (pt_map.count(zp_convert_pattern)) {
+            auto zp_convert = pt_map.at(zp_convert_pattern).get_node_shared_ptr();
+            disable_fp16_compression(zp_convert);
+        }
+
+        if (pt_map.count(scale_convert_pattern)) {
```

Comment:
Should we mark `zp_reshape_pattern` and `scale_reshape_pattern` as well?

Reply:
Yes, updated, thank you!